crates/primitives/src/transaction/pooled.rs

```diff
@@ -691,92 +688,45 @@ impl<'a> arbitrary::Arbitrary<'a> for PooledTransactionsElement {
 }
 
 /// A signed pooled transaction with recovered signer.
-#[derive(Debug, Clone, PartialEq, Eq, AsRef, Deref)]
-pub struct PooledTransactionsElementEcRecovered<T = PooledTransactionsElement> {
-    /// Signer of the transaction
-    signer: Address,
-    /// Signed transaction
-    #[deref]
-    #[as_ref]
-    transaction: T,
-}
-
-impl<T> PooledTransactionsElementEcRecovered<T> {
-    /// Create an instance from the given transaction and the [`Address`] of the signer.
-    pub const fn from_signed_transaction(transaction: T, signer: Address) -> Self {
-        Self { transaction, signer }
-    }
-
-    /// Signer of transaction recovered from signature
-    pub const fn signer(&self) -> Address {
-        self.signer
-    }
+pub type PooledTransactionsElementEcRecovered<T = PooledTransactionsElement> = RecoveredTx<T>;
```

Comment:
could we also keep an alias for the `TransactionSignedEcRecovered` but mark it as deprecated, because I assume some people are using this type via reth-primitives

Reply:
yeah we should probably do that

Reply:
also maybe we could rename `PooledTransactionsElementEcRecovered` to `RecoveredPooledTx`?

Reply:
> also maybe we could rename `PooledTransactionsElementEcRecovered` to `RecoveredPooledTx`?

I think ideally we just use `RecoveredTx<PoolTx::Pooled>`? the current `PooledTransactionsElementEcRecovered` is already not used much